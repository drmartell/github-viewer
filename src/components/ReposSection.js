import React from 'react';
import { useSelector } from 'react-redux';
import Loading from './Loading';
import { selectRepos, isReposLoading } from '../selectors/reposSelector';

const ReposSection = () => {
  const loading = useSelector(isReposLoading);
  const allRepos = useSelector(selectRepos);

  if(loading || allRepos.length === 0) return <Loading />;

  const repoElements = allRepos.map((repo, i) => <li key={i}>{repo}</li>);

  return (
    <section>
      <h1>{'Repos'}</h1>
      <ul>
        {repoElements}
      </ul>
    </section>
  );
};

export default ReposSection;
