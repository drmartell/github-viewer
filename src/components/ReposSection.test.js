import React from 'react';
import { mount } from 'enzyme';
import ReposSection from './ReposSection';
import { Provider } from 'react-redux';
import store from '../store';

describe('ReposSection component', () => {
  it('renders ReposSection', () => {
    const wrapper = mount(
      <Provider store={store}>
        <ReposSection />
      </Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
