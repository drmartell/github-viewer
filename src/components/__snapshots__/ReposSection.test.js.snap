// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposSection component renders ReposSection 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ReposSection>
    <Loading>
      <img
        src="https://mir-s3-cdn-cf.behance.net/project_modules/disp/2a6dae53616869.593a85ce553c4.gif"
        style={
          Object {
            "display": "block",
            "height": "auto",
            "margin": "auto",
            "width": "10vw",
          }
        }
      />
    </Loading>
  </ReposSection>
</Provider>
`;
